gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-rest/src/main/java/io/gravitee/rest/api/management/v2/rest/resource/integration/IntegrationResource.java

```diff
@@ -151,7 +155,7 @@ public void ingestApis(@PathParam("integrationId") String integrationId, @Suspen
     @GET
     @Path("/_preview")
     @Produces(MediaType.APPLICATION_JSON)
-    public IngestionPreviewResponse previewNewIntegrationApis(@PathParam("integrationId") String integrationId) {
+    public CompletionStage<IngestionPreviewResponse> previewNewIntegrationApis(@PathParam("integrationId") String integrationId) {
```

Comment:
I don't know if it is better but we did differently for the ingest: https://github.com/gravitee-io/gravitee-api-management/blob/master/gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-rest/src/main/java/io/gravitee/rest/api/management/v2/rest/resource/integration/IntegrationResource.java#L122